src/main/java/com/karankumar/bookproject/ui/shelf/CustomShelfForm.java

```diff
@@ -106,27 +113,56 @@ private Button createSaveButton() {
         Button save = new Button();
         save.setText("Save");
         save.addThemeVariants(ButtonVariant.LUMO_PRIMARY);
-        save.setDisableOnClick(true);
         save.addClassName("shelfFormSaveButton");
         save.addClickListener(event -> validateOnSave());
         return save;
     }
 
+    private Button createDeleteButton() {
+        delete = new Button();
+        delete.setText("Delete shelf");
+        delete.addThemeVariants(ButtonVariant.LUMO_ERROR);
+        delete.addClickListener(event -> validateOnDelete());
+        return delete;
+    }
+
     private void validateOnSave() {
         if (binder.isValid()) {
             setCustomShelfBean();
             fireEvent(new SaveEvent(this, binder.getBean()));
             closeForm();
             if (binder.getBean() != null) {
                 showSaveConfirmation();
+                owner.setEditEnabled(false);
             } else {
                 showSaveError();
             }
         }
     }
 
+    private void validateOnDelete() {
+        CustomShelf removed = this.getCustomShelf(deletingShelf);
+        fireEvent(new DeleteEvent(this, removed));
+        binder.removeBean();
+        closeForm();
+        if (binder.getBean() == null) {
+            this.showDeleteConfirmation();
```

Comment:
Is this correct to show the confirmation after closing the form?

Reply:
This confirmation shows on the main page rather than in the dialog that is closed when clicking Delete. From my testing, this is in line with the Save functionality.

Reply:
I would say its better if we can write the test along with the actual code instead of leaving it for later?